### PR TITLE
(fix #157 and other new bugs) Fix death and stagger bugs

### LINKED ISSUE #157: (fix) override stagger animation with death animation
Currently we stay in stagger animation on a killing blow, until it ends, when we will move to death animation

### PR DESCRIPTION
**Pull request by kepler471**
WE GOT OURSELVES A 3-IN-1 BUG SPLATTING COMBO HERE

### CODE REVIEW

File: Moonshot/player/player.gd [hunk 1]
```diff
@@ -135,21 +135,18 @@ func get_collider() -> CollisionShape2D:
...
 
-func _on_Player_health_depleted() -> void:
-	state_machine.transition_to("Die", {last_checkpoint = last_checkpoint})
-
-
 func set_invulnerable(time : float, animation_name = "stagger") -> void:
 	invulnerable = true
-	safety = true
+	if animation_name != "stagger":
+		safety = true
```
Comment (kepler471) [inline Moonshot/player/player.gd:141]: Stagger was turning on the safety, stopping the player from firing

File: Moonshot/player/player.gd [hunk 2]
```diff
@@ -177,23 +177,25 @@ func take_damage(damage, attack_dir, is_damage_tile: bool = false) -> void:
 			attack_dir = sign(get_global_position().x - attack_dir.x)
 			stagger_player(attack_dir,is_damage_tile)
 
-	if Utils.player_stats.health <= 0:
-		on_death()
-		
-	Utils.player_stats.take_damage(1)
```
Comment (kepler471) [inline Moonshot/player/player.gd:183]: Do not know why this was in here

File: Moonshot/player/player.gd [hunk 3]
```diff
@@ -168,7 +165,10 @@ func take_damage(damage, attack_dir, is_damage_tile: bool = false) -> void:
 	if not invulnerable:
 		Utils.player_stats.health -= damage
 		if Utils.IS_DEBUG: print("Attack Dir : ", attack_dir)
-		
+		if Utils.player_stats.health <= 0:
+			on_death()
+			safety = true
+			return
```
Comment (kepler471) [inline Moonshot/player/player.gd:171]: Precede the stagger and invuln logic with death. This disables shooting now when we die.

File: Moonshot/player/player.gd [hunk 4]
```diff
@@ -177,23 +177,25 @@ func take_damage(damage, attack_dir, is_damage_tile: bool = false) -> void:
...
-func add_health(health):
+func add_health(health) -> void:
 	var new_health = Utils.player_stats.health + health
 	Utils.player_stats.health = min(Utils.player_stats.max_health, new_health)
 
 
 func on_death() -> void:
 	CombatSignalController.emit_signal("player_kill")
 	_on_Player_health_depleted()
+	CombatSignalController.disconnect("damage_player", self, "take_damage")
```
Comment (kepler471) [inline Moonshot/player/player.gd:189]: Disconnect the signal so `take_damage()` will not be called again until game restart. The signal is connected in `_ready()`.

**Review by wattsit — APPROVED** [review]
Looks good.

**Pull request merged**